dbms/src/DataStreams/WindowTransformAction.cpp

```diff
@@ -292,6 +295,9 @@ void WindowTransformAction::initialWorkspaces()
         }
     }
 
+    // Can not have window and agg functions together in one window
+    assert((!aggregation_workspaces.empty() && !window_workspaces.empty()) == false);
```

Comment:
why have this restriction?

Reply:
> why have this restriction?

Because only agg can have this shortcut. When two boundary are all unbounded, results of window function will still be different in the same frame